storage/innobase/include/buf0buf.h

```diff
@@ -250,32 +254,16 @@ class page_id_t {
 	@return fold value */
 	inline ulint fold() const
 	{
-		/* Initialize m_fold if it has not been initialized yet. */
-		if (m_fold == ULINT_UNDEFINED) {
-			m_fold = (m_space << 20) + m_space + m_page_no;
-			ut_ad(m_fold != ULINT_UNDEFINED);
-		}
-
-		return(m_fold);
-	}
-
-	/** Copy the values from a given page_id_t object.
-	@param[in]	src	page id object whose values to fetch */
-	inline void copy_from(const page_id_t& src)
-	{
-		m_space = src.space();
-		m_page_no = src.page_no();
-		m_fold = src.fold();
+		return (m_space << 20) + m_space + m_page_no;
 	}
 
 	/** Reset the values from a (space, page_no).
 	@param[in]	space	tablespace id
 	@param[in]	page_no	page number */
 	inline void reset(ulint space, ulint page_no)
 	{
-		m_space = static_cast<ib_uint32_t>(space);
-		m_page_no = static_cast<ib_uint32_t>(page_no);
-		m_fold = ULINT_UNDEFINED;
+		m_space = static_cast<uint32_t>(space);
+		m_page_no = static_cast<uint32_t>(page_no);
```

Comment:
Do we need this member function? `fil_tablespace_iterate()` uses a combination of assignment and constructor:
```c++
 	block->page.id = page_id_t(0, 0); 
```
What does it translate into on amd64? x86? How does it compare to
```c++
	block->page.id.reset(0, 0);
```
If there is no disadvantage of using an assignment, I’d prefer it.

Reply:
> What does it translate into on amd64? x86?

It depends. There is no simple answer to your question. But here is the case where generated asm is identical. https://godbolt.org/z/diNsKS https://godbolt.org/z/hYG9hK

If will remove `reset()` for now. Thanks for noticing.